### PR TITLE
Communication delete endpoint

### LINKED ISSUE #360: Delete communications
We need an endpoint to remove communications that were for example accidentally registered.

### PR DESCRIPTION
**Pull request by MaartenS11**
This PR adds an endpoint to delete a communication. Previously this was thought to be unneeded, but it might be useful when a certain communication is registered by accident for example. This PR closes #360.

**New feature checklist**:
- [x] I've added tests for the new feature (100% coverage)
- [x] I've thoroughly documented the code
- [x] I've updated the `osoc.yaml` file

### CODE REVIEW

File: docs/osoc.yaml [hunk 1]
```diff
@@ -888,6 +888,25 @@ paths:
...
+                -   in: path
+                    required: true
+                    name: edition
+                    schema:
+                        type: string
+            responses:
+                '204':
+                    description: Returned if the removal was successful.
+                '404':
+                    description: Returned if the communication does not exist(in this edition).
```
Comment (Mikxox) [inline docs/osoc.yaml:909]: typo you forgot a space here between 'exist' and '('

File: backend/src/main/kotlin/be/osoc/team1/backend/entities/Communication.kt [hunk 2]
```diff
@@ -20,8 +21,13 @@ class Communication(
     val type: CommunicationTypeEnum,
     @JsonIgnore
     @NotBlank
-    val edition: String = ""
+    val edition: String = "",
+    @JsonIgnore
+    @OneToOne
+    val student: Student
```
Comment (Mikxox) [inline backend/src/main/kotlin/be/osoc/team1/backend/entities/Communication.kt:27]: A student can have more than one communication. Also, this field won't be send when requesting communications right (just making sure)?

**Review by Mikxox — APPROVED** [review]
This is fine for me as far as functionality goes, but I'm really not 100% up to date with backend code anymore so I might not see some stuff.

**Review by lars-vc — APPROVED** [review]

**Comment by Mikxox** [comment]
To fix the test, in PostgresConnectionTests, line 25, change
```actualDatabaseVersion shouldContain "PostgreSQL 14.2"```
to
```actualDatabaseVersion shouldContain "PostgreSQL"```
We don't really care about the version anyway
And maybe change it to 
```
@Test
    fun `when database is connected then it should be a Postgres instance running`() {
        val actualDatabase = jdbcTemplate.queryForObject("SELECT version()", String::class.java)
        actualDatabase shouldContain "PostgreSQL"
    }
```

**Pull request merged**